receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -139,20 +141,27 @@ func (mg *metricGroupPdata) toDistributionPoint(orderedLabelKeys []string, dest
 		if i != 0 {
 			adjustedCount -= mg.complexValue[i-1].value
 		}
+		// Buckets still need to be sent to know to set them as stale,
+		// but a staleness NaN converted to uint64 would be an extremely large number.
+		// Setting to 0 instead.
+		if pointIsStale {
+			adjustedCount = 0
+		}
 		bucketCounts[i] = uint64(adjustedCount)
 	}
 
 	point := dest.AppendEmpty()
 
-	if value.IsStaleNaN(mg.sum) || value.IsStaleNaN(mg.count) {
+	if pointIsStale {
 		point.SetFlags(pdataStaleFlags)
 	} else {
-		point.SetExplicitBounds(bounds)
 		point.SetCount(uint64(mg.count))
 		point.SetSum(mg.sum)
-		point.SetBucketCounts(bucketCounts)
 	}
 
+	point.SetExplicitBounds(bounds)
+	point.SetBucketCounts(bucketCounts)
```

Comment:
Do we even need to set bucket counts? IIUC, we just needed the bounds to be able to manufacture stale bucket series with NaN values later.

Reply:
It can also be done as a follow-up. I wouldn't be surprised if many exporters don't handle that case correctly.